tbot/main.py

```diff
@@ -57,11 +59,32 @@ def _import_hightlighter() -> typing.Callable[[str], str]:
 # }}}
 
 
+class TbotArgumentParser(argparse.ArgumentParser):
+    def convert_arg_line_to_args(self, arg_line: str) -> List[str]:
+        """
+        Make it possible to use shell variables also in argumentsfiles.
+        """
+        if "$" in arg_line:
+            vl = re.findall(r"(\$\{.*?\})", arg_line)
+            for v in vl:
+                varname = v[2:-1]
+                env = os.getenv(varname)
+
+                if env is None:
+                    raise RuntimeError(f"Could not find Environment variable {varname}")
+
+                # replace in arg_line
+                rp = "${" + varname + "}"
+                arg_line = arg_line.replace(rp, env)
+
+        return arg_line.split()
```

Comment:
I don't think we should do the `.split()` here.  Right now, each line of the argument file is taken as a single argument, even if it includes whitespace.  This makes the argument file robust against weird paths or other oddities.  Unless you have a particular requirement for this split, I'd like to keep things this way.

If you do require this splitting here, it would be better to use [`shlex.split()`](https://docs.python.org/3/library/shlex.html#shlex.split) which supports using `"` or `'` to prevent splitting...

Reply:
https://docs.python.org/3/library/argparse.html#customizing-file-parsing

says:
This method takes a single argument arg_line which is a string read from the argument file. It returns a list of arguments parsed from this string.

So I thought we should split(), so no real requirement from me

Reply:
Yeah the documentation is quite confusing there.  The `split()` code is just an example of a customization you could do in this function, but this isn't really explicitly pointed out...